hpedockerplugin/hpe/hpe_3par_mediator.py

```
@@ -0,0 +1,1032 @@
+                     ' the hpe3parclient.') %
+                   {'found': '.'.join(map(six.text_type, self.client_version)),
+                    'minimum': '.'.join(map(six.text_type,
+                                            MIN_CLIENT_VERSION))})
+            LOG.error(msg)
+            raise exception.HPE3ParInvalidClient(message=msg)
+
+        try:
+            self._client = file_client.HPE3ParFilePersonaClient(
+                self._config.hpe3par_api_url)
+        except Exception as e:
+            msg = (_('Failed to connect to HPE 3PAR File Persona Client: %s') %
+                   six.text_type(e))
+            LOG.exception(msg)
+            raise exception.ShareBackendException(message=msg)
+
+        try:
+            ssh_kwargs = {}
+            if self._config.hpe3par_san_ssh_port:
+                ssh_kwargs['port'] = self._config.hpe3par_san_ssh_port
+            if self._config.ssh_conn_timeout:
+                ssh_kwargs['conn_timeout'] = self._config.ssh_conn_timeout
+            if self._config.hpe3par_san_private_key:
+                ssh_kwargs['privatekey'] = \
+                    self._config.hpe3par_san_private_key
+
+            self._client.setSSHOptions(
+                self._config.hpe3par_san_ip,
+                self._config.hpe3par_san_login,
+                self._config.hpe3par_san_password,
+                **ssh_kwargs
+            )
+
+        except Exception as e:
+            msg = (_('Failed to set SSH options for HPE 3PAR File Persona '
+                     'Client: %s') % six.text_type(e))
+            LOG.exception(msg)
+            raise exception.ShareBackendException(message=msg)
+
+        LOG.info("HPE3ParMediator %(version)s, "
+                 "hpe3parclient %(client_version)s",
+                 {"version": self.VERSION,
+                  "client_version": hpe3parclient.get_version_string()})
+
+        try:
+            wsapi_version = self._client.getWsApiVersion()['build']
```

@imran-ansari will not it be good if we check here whether wsapi_version is less than required wsapi version.
If this is non supported version can we throw an error?

---

WSAPI check is required for 3.3.1 Minimum since we are using some REST API's specific for 3.3.1